lib/rmt/misc.rb

```diff
@@ -4,6 +4,9 @@ module RMT
   module Misc
     def self.make_repo_url(base_url, local_path, service_name = nil)
       uri = URI.join(base_url, File.join(RMT::DEFAULT_MIRROR_URL_PREFIX, local_path))
+      # NOTE: Make sure to only add the credentials where necessary (Pubcloud? or SMT?)
+      # In all other cases do not add them, since this will break other repository
+      # managers such as yum!
```

Comment:
Could we make this conditional, so that the credentials parameter is only skipped if `StrictAuthentication::Engine` is not enabled?

Reply:
I dont think that is the/an issue on the Public Cloud
as AFAIU it gets overwritten by the [zypper-auth method](https://github.com/SUSE/rmt/blob/024478b7c8a9b7c45d617377f2c472917220d603/engines/zypper_auth/lib/zypper_auth/engine.rb#L113)

testing the code as it is on this PR on RMT
```bash
tux> registercloudguest
Registration succeeded
tux> zypper ls -u
...
9 | Web_and_Scripting_Module_x86_64     | Web_and_Scripting_Module_x86_64     | Yes     | ----      | Yes     | ris  | plugin:/susecloud?credentials=Web_and_Scripting_Module_x86_64&path=/services/2494
```
so credentials are present

the `credentials` on the zypper-auth engine side need to be present on the Public Cloud case

Reply:
I looked into it. https://github.com/SUSE/rmt/blob/master/engines/zypper_auth/lib/zypper_auth/engine.rb#L111 is overwriting the implementation, drops potential `?credentials=` parameter and then later adds them manually again. This works fine and no double `?credentials` since it only uses, `url.path`..